jerry-core/jerry-api.h

```diff
@@ -147,71 +173,83 @@ jerry_value_t jerry_value_to_string (const jerry_value_t);
+jerry_value_t jerry_create_external_function (jerry_external_handler_t);
+jerry_value_t jerry_create_number (double);
+jerry_value_t jerry_create_null (void);
+jerry_value_t jerry_create_object (void);
+jerry_value_t jerry_create_string (const jerry_char_t *);
+jerry_value_t jerry_create_string_sz (const jerry_char_t *, jerry_size_t);
+jerry_value_t jerry_create_undefined (void);
 
 /**
  * Functions of array objects
  */
-bool jerry_set_array_index_value (jerry_object_t *, jerry_length_t, jerry_value_t);
-bool jerry_get_array_index_value (jerry_object_t *, jerry_length_t, jerry_value_t *);
-uint32_t jerry_get_array_length (const jerry_object_t *);
+jerry_value_t jerry_set_array_index_value (const jerry_value_t, uint32_t, const jerry_value_t);
+jerry_value_t jerry_get_array_index_value (const jerry_value_t , uint32_t);
+uint32_t jerry_get_array_length (const jerry_value_t);
 
 /**
  * Functions of 'jerry_string_t'
  */
-jerry_size_t jerry_get_string_size (const jerry_string_t *);
-jerry_length_t jerry_get_string_length (const jerry_string_t *);
-jerry_size_t jerry_string_to_char_buffer (const jerry_string_t *, jerry_char_t *, jerry_size_t);
+jerry_size_t jerry_get_string_size (const jerry_value_t);
+jerry_length_t jerry_get_string_length (const jerry_value_t);
+jerry_size_t jerry_string_to_char_buffer (const jerry_value_t, jerry_char_t *, jerry_size_t);
 
 /**
  * General API functions of JS objects
  */
-bool jerry_is_array (const jerry_object_t *);
-bool jerry_is_constructor (const jerry_object_t *);
-bool jerry_is_function (const jerry_object_t *);
-bool jerry_add_object_field (jerry_object_t *, const jerry_char_t *, jerry_size_t, const jerry_value_t, bool);
-bool jerry_delete_object_field (jerry_object_t *, const jerry_char_t *, jerry_size_t);
-jerry_value_t jerry_get_object_field_value (jerry_object_t *, const jerry_char_t *);
-jerry_value_t jerry_get_object_field_value_sz (jerry_object_t *, const jerry_char_t *, jerry_size_t);
-bool jerry_set_object_field_value (jerry_object_t *, const jerry_char_t *, const jerry_value_t);
-bool jerry_set_object_field_value_sz (jerry_object_t *, const jerry_char_t *, jerry_size_t, const jerry_value_t);
-bool jerry_foreach_object_field (jerry_object_t *, jerry_object_field_foreach_t, void *);
-bool jerry_get_object_native_handle (jerry_object_t *, uintptr_t *);
-void jerry_set_object_native_handle (jerry_object_t *, uintptr_t, jerry_object_free_callback_t);
-jerry_value_t jerry_construct_object (jerry_object_t *, const jerry_value_t[], uint16_t);
-jerry_value_t jerry_call_function (jerry_object_t *, jerry_object_t *, const jerry_value_t[], uint16_t);
+bool jerry_has_property (const jerry_value_t, const jerry_value_t);
+bool jerry_has_own_property (const jerry_value_t, const jerry_value_t);
+bool jerry_delete_property (const jerry_value_t, const jerry_value_t);
+
+jerry_value_t jerry_set_data_property (const jerry_value_t, const jerry_value_t, const jerry_value_t);
```

Comment:
Does it set data property only? I mean, if a set accessor is defined, does it not call this accessor? If it works with accessors as well, I would call it jerry_set_property_value